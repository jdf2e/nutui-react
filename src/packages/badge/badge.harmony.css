.nut-badge {
  position: relative;
  display: inline-block;
}
.nut-badge-icon {
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  background: linear-gradient(135deg, #f53d4d 0%, #fa2c19 100%);
  top: -10%;
  right: -15%;
  padding: 3px;
  text-align: center;
  border: 0px solid #ffffff;
  border-radius: 14px;
  z-index: 1;
}
.nut-badge-icon .nut-icon {
  width: 12px;
  height: 12px;
  font-size: 12px;
}
.nut-badge-sup {
  height: 14px;
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  background: linear-gradient(135deg, #f53d4d 0%, #fa2c19 100%);
  color: #fff;
  padding: 0 4px;
  font-size: 10px;
  font-weight: normal;
  text-align: center;
  border: 0px solid #ffffff;
  border-radius: 14px;
  min-width: 5px;
  white-space: nowrap;
  z-index: 1;
}
.nut-badge-one {
  height: 14px;
  width: 14px;
  padding: 0;
}
.nut-badge-content {
  transform: translateY(-50%) translateX(100%);
}
.nut-badge-dot {
  width: 7px;
  height: 7px;
  border: 0px solid #ffffff;
  border-radius: 7px;
  padding: 0;
}

[dir=rtl] .nut-badge-icon,
.nut-rtl .nut-badge-icon {
  right: auto;
  left: -15%;
}
[dir=rtl] .nut-badge-content {
  transform: translateY(-50%) translateX(-100%);
}
.nut-rtl .nut-badge-content {
  transform: translateY(-50%) translateX(-100%);
}