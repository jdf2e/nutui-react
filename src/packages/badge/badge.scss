.nut-badge {
  position: relative;
  display: inline-block;

  &-icon {
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
    background: $badge-background-color;
    top: -10%;
    right: -15%;
    padding: $badge-icon-padding;
    text-align: center;
    border: $badge-border;
    border-radius: $badge-border-radius;
    z-index: $badge-z-index;

    .nut-icon {
      width: $badge-icon-size;
      height: $badge-icon-size;
      font-size: $badge-icon-size;
    }
  }

  &-sup {
    height: $badge-height;
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
    background: $badge-background-color;
    color: $badge-color;
    padding: $badge-padding;
    font-size: $badge-font-size;
    font-weight: normal;
    text-align: center;
    border: $badge-border;
    border-radius: $badge-border-radius;
    min-width: $badge-min-width;
  }

  &-one {
    height: $badge-height;
    width: $badge-height;
    padding: 0;
  }

  &-content {
    transform: $badge-content-transform;
  }

  &-dot {
    width: $badge-dot-width;
    height: $badge-dot-width;
    border: $badge-border;
    border-radius: $badge-dot-width;
    padding: 0;
  }
}

[dir='rtl'] .nut-badge {
  &-icon {
    right: auto;
    left: -15%;
  }
}
