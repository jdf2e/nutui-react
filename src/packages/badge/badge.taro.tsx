import React, { CSSProperties, FunctionComponent, ReactNode } from 'react'
import classNames from 'classnames'

import { BasicComponent, ComponentDefaults } from '@/utils/typings'

export interface BadgeProps extends BasicComponent {
  value: string | ReactNode
  dot: boolean
  max: number
  top: string
  right: string
  color: string
  children?: ReactNode
}
export type BadgeType = 'default' | 'primary' | 'success' | 'warning' | 'danger'
const defaultProps = {
  ...ComponentDefaults,
  value: '',
  dot: false,
  max: 99,
  top: '0',
  right: '5',
  color: '',
} as BadgeProps
export const Badge: FunctionComponent<Partial<BadgeProps>> = (props) => {
  const { className, style, value, max, children, dot, top, right, color } = {
    ...defaultProps,
    ...props,
  }
  const classPrefix = 'nut-badge'
  const classes = classNames(classPrefix, className)
  const contentClasses = classNames(
    { [`${classPrefix}__dot`]: dot },
    `${classPrefix}__content`,
    `${classPrefix}__sup`
  )
  function content() {
    if (dot || typeof value === 'object') return null
    if (typeof value === 'number' && typeof max === 'number') {
      return max < value ? `${max}+` : value
    }
    return value
  }
  const getStyle = () => {
    const style: CSSProperties = {}
    style.top = `${Number(top) || parseFloat(top) || 0}px`
    style.right = `${Number(right) || parseFloat(right) || 0}px`
    style.background = color
    return style
  }
  return (
    <div className={classes} style={style}>
      {typeof value === 'object' && value && (
        <div className={`${classPrefix}__icon`}>{value}</div>
      )}
      <div>{children}</div>
      <div className={contentClasses} style={getStyle()}>
        {content()}
      </div>
    </div>
  )
}

Badge.defaultProps = defaultProps
Badge.displayName = 'NutBadge'
