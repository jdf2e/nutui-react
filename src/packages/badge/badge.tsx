import React, { CSSProperties, FunctionComponent, ReactNode } from 'react'

import Icon from '@/packages/icon'

import { BasicComponent, ComponentDefaults } from '@/utils/typings'

export interface BadgeProps extends BasicComponent {
  value: any
  dot: boolean
  max: number
  top: string
  right: string
  zIndex: string
  color: string
  icons: any
  children?: ReactNode
}

export type BadgeType = 'default' | 'primary' | 'success' | 'warning' | 'danger'

const defaultProps = {
  ...ComponentDefaults,
  value: '',
  dot: false,
  max: 10000,
  top: '0',
  right: '0',
  zIndex: '0',
  color: '',
  icons: '',
} as BadgeProps
export const Badge: FunctionComponent<Partial<BadgeProps>> = (props) => {
  const {
    className,
    style,
    children,
    dot,
    top,
    right,
    zIndex,
    color,
    icons,
    iconClassPrefix,
    iconFontClassName,
  } = {
    ...defaultProps,
    ...props,
  }
  function content() {
    if (dot) return undefined
    const { value } = props
    const { max } = props
    if (typeof value === 'number' && typeof max === 'number') {
      return max < value ? `${max}+` : value
    }
    return value
  }
  const getStyle = () => {
    const style: CSSProperties = {}
    style.top = `${Number(top) || parseFloat(top) || 0}px`
    style.right = `${Number(right) || parseFloat(right) || 0}px`
    style.zIndex = zIndex
    style.backgroundColor = color
    return style
  }
  return (
    <div className={`nut-badge ${className}`} style={style}>
      {icons !== '' && (
        <div className="slot-icons">
          <Icon
            classPrefix={iconClassPrefix}
            fontClassName={iconFontClassName}
            className="_icon"
            name={icons}
            color="#ffffff"
            size="12"
          />
        </div>
      )}
      <div>{children}</div>
      <div
        className={`${dot ? 'is-dot' : ''} nut-badge__content sup`}
        style={getStyle()}
      >
        {content()}
      </div>
    </div>
  )
}

Badge.defaultProps = defaultProps
Badge.displayName = 'NutBadge'
