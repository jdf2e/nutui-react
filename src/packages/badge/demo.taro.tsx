import React from 'react'
import { useTranslate } from '@/sites/assets/locale/taro'
import { Avatar, Badge, Cell, CellGroup } from '@/packages/nutui.react.taro'
interface T {
  '8ab98966': string
  '1e7a2282': string
  '781b07fd': string
  '1c730245': string
  '915d7b01': string
  f1089312: string
}
const BadgeDemo = () => {
  const [translated] = useTranslate<T>({
    'zh-CN': {
      '8ab98966': '默认用法',
      '1e7a2282': '最大值',
      '781b07fd': '自定义颜色',
      '1c730245': '自定义徽标内容',
      '915d7b01': '自定义位置',
      f1089312: '独立展示',
    },
    'zh-TW': {
      '8ab98966': '默认用法',
      '1e7a2282': '最大值',
      '781b07fd': '自定义颜色',
      '1c730245': '自定义徽标内容',
      '915d7b01': '自定义位置',
      f1089312: '独立展示',
    },
    'en-US': {
      '8ab98966': '默认用法',
      '1e7a2282': '最大值',
      '781b07fd': '自定义颜色',
      '1c730245': '自定义徽标内容',
      '915d7b01': '自定义位置',
      f1089312: '独立展示',
    },
  })
  return (
    <>
      <div className="demo">
        <h2>{translated['8ab98966']}</h2>
        <CellGroup>
          <Cell>
            <Badge value={8}>
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value={76}>
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value="NEW">
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge dot>
              <Avatar icon="my" shape="square" />
            </Badge>
          </Cell>
        </CellGroup>

        <h2>{translated['1e7a2282']}</h2>
        <CellGroup>
          <Cell>
            <Badge value={200} max={9}>
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value={200} max={20}>
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value={200} max={99}>
              <Avatar icon="my" shape="square" />
            </Badge>
          </Cell>
        </CellGroup>

        <h2>{translated['781b07fd']}</h2>
        <CellGroup>
          <Cell>
            <Badge
              value={8}
              color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
            >
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge
              value={76}
              color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
            >
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge
              value="NEW"
              color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
            >
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge
              dot
              color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
            >
              <Avatar icon="my" shape="square" />
            </Badge>
          </Cell>
        </CellGroup>

        <h2>{translated['1c730245']}</h2>
        <CellGroup>
          <Cell>
            <Badge icons="checklist">
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge icons="link">
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge icons="download">
              <Avatar icon="my" shape="square" />
            </Badge>
          </Cell>
        </CellGroup>

        <h2>{translated['915d7b01']}</h2>
        <CellGroup>
          <Cell>
            <Badge value={8} top="5" right="5">
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value={76} top="10" right="10">
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value="NEW">
              <Avatar icon="my" shape="square" />
            </Badge>
          </Cell>
        </CellGroup>

        <h2>{translated.f1089312}</h2>
        <CellGroup>
          <Cell>
            <Badge value={8} />
            <Badge value={76} />
            <Badge value="NEW" />
          </Cell>
        </CellGroup>
      </div>
    </>
  )
}

export default BadgeDemo
