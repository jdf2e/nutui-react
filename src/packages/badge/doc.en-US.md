# Badge

## Intro

出现在图标或文字右上角的红色圆点、数字或者文字，表示有新内容或者待处理的信息。

## Install

```tsx
import { Badge } from '@nutui/nutui-react';
```

## Demo

### Basic usage

:::demo

```tsx
import React from "react";
import { Badge, Avatar, Cell } from '@nutui/nutui-react';
import { My } from '@nutui/icons-react';

const App = () => {
  return (
    <Cell>
      <Badge value={8}>
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge value={76}>
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge value="NEW">
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge dot>
        <Avatar icon={<My />} shape="square" />
      </Badge>
    </Cell>
  )
}
export default App;
```

:::

### Max Size

:::demo

```tsx
import React from "react";
import { Badge, Avatar, Cell } from '@nutui/nutui-react';
import { My } from '@nutui/icons-react';

const App = () => {
  return (
    <Cell>
      <Badge value={200} max={9}>
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge value={200} max={20}>
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge value={200} max={99}>
        <Avatar icon={<My />} shape="square" />
      </Badge>
    </Cell>
  )
}
export default App;
```

:::

### Custom Color

:::demo

```tsx
import React from "react";
import { Badge, Avatar, Cell } from '@nutui/nutui-react';
import { My } from '@nutui/icons-react';

const App = () => {
  return (
    <Cell>
      <Badge
        value={8}
        color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
      >
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge
        value={76}
        color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
      >
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge
        value="NEW"
        color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
      >
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge
        dot
        color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
      >
        <Avatar icon={<My />} shape="square" />
      </Badge>
    </Cell>
  )
}
export default App;
```

:::

### Custom context

:::demo

```tsx
import React from "react";
import { Badge, Avatar, Cell } from '@nutui/nutui-react';
import { My, Checklist, Link as LinkIcon, Download } from '@nutui/icons-react';

const App = () => {
  return (
    <Cell>
      <Badge
        value={<Checklist color="#fff" width={12} height={12} />}
        className="test"
      >
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge value={<LinkIcon color="#fff" width={12} height={12} />}>
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge value={<Download color="#fff" width={12} height={12} />}>
        <Avatar icon={<My />} shape="square" />
      </Badge>
    </Cell>
  )
}
export default App;
```

:::

### Custom CSS

:::demo

```tsx
import React from "react";
import { Badge, Avatar, ConfigProvider, Cell } from '@nutui/nutui-react';
import { My } from '@nutui/icons-react';

const customTheme = {
  nutuiBadgeBorderRadius: '12px 12px 12px 0',
}

const customTheme2 = {
  nutuiBadgeDotWidth: '14px',
  nutuiBadgeDotHeight: '14px',
  nutuiBadgeBorder: '2px solid #fff',
}

const App = () => {
  return (
    <Cell>
      <ConfigProvider theme={customTheme}>
        <Badge value="NEW">
          <Avatar icon={<My />} shape="square" />
        </Badge>
      </ConfigProvider>

      <ConfigProvider theme={customTheme2}>
        <Badge dot top="2" right="8">
          <Avatar icon={<My />} shape="square" />
        </Badge>
      </ConfigProvider>
    </Cell>
  )
}
export default App;
```

:::

### Custom Position

:::demo

```tsx
import React from "react";
import { Badge, Avatar, Cell } from '@nutui/nutui-react';
import { My } from '@nutui/icons-react';

const App = () => {
  return (
    <Cell>
      <Badge value={8} top="5" right="5">
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge value={76} top="10" right="10">
        <Avatar icon={<My />} shape="square" />
      </Badge>
      <Badge value="NEW">
        <Avatar icon={<My />} shape="square" />
      </Badge>
    </Cell>
  )
}
export default App;
```

:::

### Display Alone

:::demo

```tsx
import React from "react";
import { Badge,Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell style={{height: '100px'}}>
      <Badge value={8}> </Badge>
      <Badge value={76}> </Badge>
      <Badge value="NEW"> </Badge>
    </Cell>
  )
}
export default App;
```

:::

## Badge

### Props

| Property | Description | Type | Default |
| --- | --- | --- | --- |
| value | value to show, eg number、charctor and custom content | `ReactNode` | `-` |
| max | when value is number, it's the max size | `number` | `99` |
| dot | Is dotted | `boolean` | `false` |
| top | Up and down offset, support unit setting, can be set to: "5", etc. | `string` | `"0"` |
| right | Left and right offset, support unit setting, can be set to: "5", etc. | `string` | `"5"` |
| color | background color,the default value is the theme primary color | `string` | `-` |

## Theming

### CSS Variables

The component provides the following CSS variables, which can be used to customize styles. Please refer to [ConfigProvider component](#/en-US/component/configprovider).

| Name | Description | Default |
| --- | --- | --- |
| \--nutui-badge-background-color | badge background color | `linear-gradient(135deg, $primary-color 0%, $primary-color-end 100%))` |
| \--nutui-badge-color | badge content color | `#fff` |
| \--nutui-badge-font-size | badge content font size | `$font-size-1` |
| \--nutui-badge-border | badge border | `0px solid $primary-text-color` |
| \--nutui-badge-border-radius | badge border-radius | `14px` |
| \--nutui-badge-padding | badge padding value | `0 5px` |
| \--nutui-badge-icon-padding | when badge is icon,badge padding | `2px` |
| \--nutui-badge-content-transform | badge content transform | `translateY(-50%) translateX(100%)` |
| \--nutui-badge-z-index | when badge is icon, badge z-index | `1` |
| \--nutui-badge-dot-width | when badge is dot, the dot width,height and border radius | `7px` |