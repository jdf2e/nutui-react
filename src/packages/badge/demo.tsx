import React from 'react'
import { Badge } from './badge'
import Cell from '@/packages/cell'
import CellGroup from '@/packages/cellgroup'
import Avatar from '@/packages/avatar'

const BadgeDemo = () => {
  return (
    <>
      <div className="demo">
        <h2>默认用法</h2>
        <CellGroup>
          <Cell>
            <Badge value={8}>
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value={76}>
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value="NEW">
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge dot>
              <Avatar icon="my" shape="square" />
            </Badge>
          </Cell>
        </CellGroup>

        <h2>最大值</h2>
        <CellGroup>
          <Cell>
            <Badge value={200} max={9}>
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value={200} max={20}>
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value={200} max={99}>
              <Avatar icon="my" shape="square" />
            </Badge>
          </Cell>
        </CellGroup>

        <h2>自定义颜色</h2>
        <CellGroup>
          <Cell>
            <Badge
              value={8}
              color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
            >
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge
              value={76}
              color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
            >
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge
              value="NEW"
              color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
            >
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge
              dot
              color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
            >
              <Avatar icon="my" shape="square" />
            </Badge>
          </Cell>
        </CellGroup>

        <h2>自定义徽标内容</h2>
        <CellGroup>
          <Cell>
            <Badge icons="checklist">
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge icons="link">
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge icons="download">
              <Avatar icon="my" shape="square" />
            </Badge>
          </Cell>
        </CellGroup>

        <h2>自定义位置</h2>
        <CellGroup>
          <Cell>
            <Badge value={8} top="5" right="5">
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value={76} top="10" right="10">
              <Avatar icon="my" shape="square" />
            </Badge>
            <Badge value="NEW">
              <Avatar icon="my" shape="square" />
            </Badge>
          </Cell>
        </CellGroup>

        <h2>独立展示</h2>
        <CellGroup>
          <Cell>
            <Badge value={8} />
            <Badge value={76} />
            <Badge value="NEW" />
          </Cell>
        </CellGroup>
      </div>
    </>
  )
}

export default BadgeDemo
