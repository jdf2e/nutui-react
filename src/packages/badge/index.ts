import { Badge } from './badge'

export type { BadgeProps } from './badge'
export default Badge
