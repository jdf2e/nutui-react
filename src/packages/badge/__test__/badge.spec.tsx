import * as React from 'react'
import { render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'

import { Badge } from '../badge'
import Button from '../../button'

test('should match snapshot', () => {
  const { asFragment } = render(<Badge value={8} />)
  expect(asFragment()).toMatchSnapshot()
})

test('should match max size', () => {
  const { asFragment } = render(<Badge value={200} max={9} />)
  expect(asFragment()).toMatchSnapshot()
})

test('should match dot', () => {
  const { asFragment } = render(<Badge value={10} dot />)
  expect(asFragment()).toMatchSnapshot()
})

test('should match top、right: bad number', () => {
  const { asFragment } = render(<Badge value={10} top="--10" right="0" />)
  expect(asFragment()).toMatchSnapshot()
})

test('should match top、right: have px', () => {
  const { asFragment } = render(<Badge value={10} top="-10px" right="0" />)
  expect(asFragment()).toMatchSnapshot()
})

test('should match top、right: float', () => {
  const { asFragment } = render(
    <Badge value={10} top="-10.8px" right="0.5px" />
  )
  expect(asFragment()).toMatchSnapshot()
})

test('should match custom color', () => {
  const { asFragment } = render(<Badge value={200} color="orange" />)
  expect(asFragment()).toMatchSnapshot()
})

test('should match custom gradient color', () => {
  const { asFragment } = render(
    <Badge
      value={200}
      color="linear-gradient(315deg, rgba(73,143,242,1) 0%,rgba(73,101,242,1) 100%)"
    />
  )
  expect(asFragment()).toMatchSnapshot()
})

test('should match custom icons', () => {
  const { asFragment } = render(<Badge value={200} icons="link" />)
  expect(asFragment()).toMatchSnapshot()
})
