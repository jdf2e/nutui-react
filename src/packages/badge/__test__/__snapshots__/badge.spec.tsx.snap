// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match custom icon 1`] = `
<DocumentFragment>
  <div
    class="nut-badge"
  >
    <div
      class="nut-badge-icon"
    >
      <svg
        aria-labelledby="Checklist"
        class="nut-icon nut-icon-Checklist "
        role="presentation"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M159.29 500.62c62.58 0 125.16 17.07 221.87 102.4 8.53 5.69 19.91 5.69 25.6 0 48.35-54.04 238.93-261.69 455.11-329.95 0 0 28.45-5.69 42.66 19.91 8.53 17.07 19.91 34.13-5.69 54.04-22.75 17.07-264.53 179.2-440.88 440.89l-2.85 2.85c-11.38 8.53-68.27 51.2-119.47-14.23-56.89-71.11-85.33-139.38-196.26-196.26-2.85 0-2.85-2.85-5.69-5.69-11.38-11.38-54.04-73.95 25.6-73.96z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nut-badge"
  >
    <div
      class="nut-badge-content nut-badge-sup nut-badge-one"
      style="top: 0px; right: 0px;"
    >
      8
    </div>
  </div>
</DocumentFragment>
`;
